### PR TITLE
fix(pkg): open rev lock with same settings as dune lock

### PR DESCRIPTION
**Pull request by rgrinberg**
It would be better to share the code, but this is good enough for now.

### CODE REVIEW

File: src/dune_pkg/rev_store.ml [hunk 1]
```diff
@@ -33,7 +33,13 @@ let with_flock lock_path ~f =
...
   let parent = Path.parent_exn lock_path in
   Path.mkdir_p parent;
-  let fd = Unix.openfile (Path.to_string lock_path) [ Unix.O_CREAT; O_RDONLY ] 0o644 in
+  let fd =
+    Unix.openfile
+      (Path.to_string lock_path)
+      [ Unix.O_CREAT; O_RDONLY; O_SHARE_DELETE ]
+      0o600
+  in
+  Unix.set_close_on_exec fd;
```
Comment (Leonidas-from-XIV) [inline src/dune_pkg/rev_store.ml:42]: I think `O_CLOEXEC` would be better in that case, since this also avoids potential race conditions.

**Review by Leonidas-from-XIV — COMMENTED** [review]
Any reason why these flags & modes in particular were chosen?

File: src/dune_pkg/rev_store.ml (hunk 1 above)
Comment (rgrinberg) [reply src/dune_pkg/rev_store.ml]: ah yes indeed.

**Comment by rgrinberg** [comment]
> Any reason why these flags & modes in particular were chosen?

The mysterious flag is `O_SHARE_DELETE` and I think @nojb added it for Windows

**Comment by nojb** [comment]
> > Any reason why these flags & modes in particular were chosen?
> 
> The mysterious flag is `O_SHARE_DELETE` and I think @nojb added it for Windows

Indeed, this flag is needed so that the file can be deleted even if it is open, matching the usual semantics on Unix.

**Review by Leonidas-from-XIV — APPROVED** [review]

**Pull request merged**